salt/addons/init.sls

```diff
@@ -14,40 +10,44 @@ include:
     - source:      salt://addons/addons/namespace.yaml.jinja
     - template:    jinja
 
-{% if salt['pillar.get']('addons:dns', 'false').lower() == 'true' %}
-/etc/kubernetes/addons/kubedns-sa.yaml:
-  file.managed:
-    - source:     salt://addons/addons/kubedns-sa.yaml.jinja
-    - template:   jinja
-
-/etc/kubernetes/addons/kubedns-cm.yaml:
-  file.managed:
-    - source:      salt://addons/addons/kubedns-cm.yaml.jinja
-    - template:    jinja
+apply-namespace:
+  cmd.run:
+    - name: |
+        kubectl apply -f /etc/kubernetes/addons/namespace.yaml
```

Comment:
We knew this would happen, but I don't think running `kubectl apply` twice could be considered safe. The only safe solution would be to run this only once in the cluster, ideally it in the Salt master with the k8s Salt module...